### ISSUE TITLE
PyPy Support?

### ISSUE THREAD

**Issue by jakirkham** [cupy/cupy#3221]
Does CuPy support PyPy? Are there any changes users need to make to ensure a build works with PyPy? If this isn't supported, is there a sense of how much work would be required here?

**Comment by leofang** [comment]
ref: https://github.com/conda-forge/cupy-feedstock/issues/43#issuecomment-603032749

`import cupy` is OK, but there's a `CUDA_ERROR_INVALID_IMAGE` driver error when trying to run CuPy with PyPy.

**Comment by leofang** [comment]
I think this issue can be closed. I tested the backport PR #3227 with v7.3.0 and it worked (importable, runnable, and passed a few randomly picked tests). 

I did see the test `TestCumprod.test_cumprod_huge_array` failed, though. I got `OutOfMemoryError` for this. But we should open a separate issue to follow up. Most likely it involves how PyPy does garbage collection, which I assume must be quite different from CPython's.
